tests/unittests/tests-credman/tests-credman.c

```diff
@@ -0,0 +1,247 @@
+    ret = credman_add(&in_credential);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+    TEST_ASSERT_EQUAL_INT(++exp_count, credman_get_used_count());
+
+    /* delete a credential from system buffer */
+    ret = credman_delete(in_credential.tag, in_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, ret);
+    TEST_ASSERT_EQUAL_INT(--exp_count, credman_get_used_count());
+
+    /* get the deleted credential */
+    ret = credman_get(&out_credential, in_credential.tag, in_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NOT_FOUND, ret);
+
+    /* delete a deleted credential */
+    ret = credman_delete(in_credential.tag, in_credential.type);
+    TEST_ASSERT_EQUAL_INT(CREDMAN_NOT_FOUND, ret);
+    TEST_ASSERT_EQUAL_INT(exp_count, credman_get_used_count());
+}
+
+static void test_credman_delete_random_order(void)
+{
+    credman_tag_t tag1 = CREDMAN_TEST_TAG;
+    credman_tag_t tag2 = CREDMAN_TEST_TAG + 1;
+
+    credman_credential_t out_credential;
+    credman_credential_t in_credential = {
+        .tag = tag1,
+        .type = CREDMAN_TYPE_ECDSA,
+        .params = {
+            .ecdsa = {
+                .private_key = ecdsa_priv_key,
+                .public_key = { .x = ecdsa_pub_key_x, .y = ecdsa_pub_key_y },
+                .client_keys = NULL,
+                .client_keys_size = 0,
+            },
+        },
+    };
+
+    // fill the system pool, assume CREDMAN_MAX_CREDENTIALS is 2
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_add(&in_credential));
+    in_credential.tag = tag2;
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_add(&in_credential));
+
+    // delete the first credential
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_delete(tag1, in_credential.type));
+
+    // get the second credential
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_get(&out_credential, tag2, in_credential.type));
+    TEST_ASSERT(!_compare_credentials(&in_credential, &out_credential));
+}
+
+static void test_credman_add_delete_all(void)
+{
+    credman_tag_t tag1 = CREDMAN_TEST_TAG;
+    credman_tag_t tag2 = CREDMAN_TEST_TAG + 1;
+
+    credman_credential_t in_credential = {
+        .tag = tag1,
+        .type = CREDMAN_TYPE_ECDSA,
+        .params = {
+            .ecdsa = {
+                .private_key = ecdsa_priv_key,
+                .public_key = { .x = ecdsa_pub_key_x, .y = ecdsa_pub_key_y },
+                .client_keys = NULL,
+                .client_keys_size = 0,
+            },
+        },
+    };
+
+    // add credentials
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_add(&in_credential));
+    in_credential.tag = tag2;
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_add(&in_credential));
+    TEST_ASSERT_EQUAL_INT(2, credman_get_used_count());
+
+    // delete starting from first added credential
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_delete(tag1, in_credential.type));
+    TEST_ASSERT_EQUAL_INT(CREDMAN_OK, credman_delete(tag2, in_credential.type));
+    TEST_ASSERT_EQUAL_INT(0, credman_get_used_count());
```

Comment:
Please also add a test that tries adding stuff after space what made through deletion